src/main/java/seedu/duke/command/ListNoteCommand.java

```diff
@@ -85,61 +90,80 @@ public ListNoteCommand(Boolean isAscendingOrder, ArrayList<String> tags) {
      */
     @Override
     public String execute() {
-        String noteString = "";
+        StringBuilder noteString = new StringBuilder();
         ArrayList<Note> notes = new ArrayList<>();
 
         ArrayList<Note> sortedNotes = (ArrayList<Note>) notebook.getNotes().stream()
-                .filter((s) -> s instanceof Note)
-                .sorted((a, b) -> a.getTitle().toLowerCase().compareTo(b.getTitle().toLowerCase()))
+                .filter(Objects::nonNull)
+                .sorted(Comparator.comparing(a -> a.getTitle().toLowerCase()))
                 .collect(Collectors.toList());
 
         if (tags == null) {
 
             if (isAscendingOrder == null) {
                 for (int i = 0; i < notebook.getNotes().size(); i++) {
-                    noteString += (i + 1) + "." + notebook.getNotes().get(i).toString();
+                    noteString.append(i + 1).append(".")
+                            .append(notebook.getNotes().get(i).getTitle())
+                            .append(InterfaceManager.LS);
                 }
             } else {
-                noteString = getSortedString(noteString, sortedNotes);
+                noteString = new StringBuilder(getSortedString(noteString.toString(), sortedNotes));
             }
-        } else {
-            Map<Tag, ArrayList<Note>> tag = tagManager.getTagMap();
-
-            // Based on user inputted tags, will store the respective values in an ArrayList
-            // E.g. if user input 2 tags, CS2113 and important, will have 2 ArrayList
-            //      1 for the values corresponding to CS2113 and the other for important tag
-            List<ArrayList<Note>> values = tags.stream()
-                    .map(tag::get)
-                    .collect(Collectors.toList());
-
-            for (int i = 0; i < values.size(); i++) {
-                for (int j = 0; j < values.get(i).size(); j++) {
-                    // Account for duplicates.
-                    // In case an item has both CS2113 and Important tag
-                    if (!notes.contains(values.get(i).get(j))) {
-                        notes.add(values.get(i).get(j));
-                    }
+
+            return noteString.toString();
+        }
+
+        // Obtaining ArrayList<String> of tags and parsing it to get an ArrayList<Tag> of tags
+        Map<Tag, ArrayList<Note>> tagMap = tagManager.getTagMap();
+        ArrayList<Tag> tagList = new ArrayList<>();
+
+        for (String tag : tags) {
+            Tag currentTag = tagManager.getTag(tag);
+
+            if (tag != null) {
+                tagList.add(currentTag);
+            }
+        }
+
+        // Based on user inputted tags, will store the respective values in an ArrayList
+        // E.g. if user input 2 tags, CS2113 and important, will have 2 ArrayList
+        //      1 for the values corresponding to CS2113 and the other for important tag
```

Comment:
```suggestion
        /**
          * Based on user inputted tags, will store the respective values in an ArrayList
          * E.g. if user input 2 tags, CS2113 and important, will have 2 ArrayList
          * 1 for the values corresponding to CS2113 and the other for important tag
          */
```

Reply:
oops messed up the block comment.

Reply:
Correct me if I am wrong but I don't think JavaDoc works inside function? Like it won't be displayed on the html.

Reply:
Yup doesn't work. So will not be making this change